### PR TITLE
Update Project.toml

### PR DESCRIPTION
**Pull request by OkonSamuel**
This **PR** updates the projects **Project.toml** file for compatibility with **Julia 1** and **Julia 1.4**. It also fixes some deprecations. Compatibility with **Julia 0.7** is no longer supported.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,14 +3,12 @@ os:
     - linux
     - osx
 julia:
-    - 0.7
-    - '1.0'
+    - 1.0
```
Comment (cstjean) [inline .travis.yml]: We should also test on `1.4`

**Comment by cstjean** [comment]
Thanks a lot! Do you know why the tests didn't run? I also had that on my other package.

**Comment by cstjean** [comment]
Still no CI triggered 😢 Should I merge anyway? Can you run the tests locally?

**Comment by OkonSamuel** [comment]
> Still no CI triggered 😢 Should I merge anyway? Can you run the tests locally?

The tests are running but they aren't showing under the github repo. Check your travisci repo [page](https://travis-ci.org/github/cstjean/ScikitLearn.jl/jobs/667519138?utm_medium=notification&utm_source=github_status). The build is failing there.

**Comment by OkonSamuel** [comment]
@cstjean Still hold off on the merging.

**Comment by OkonSamuel** [comment]
> > Still no CI triggered 😢 Should I merge anyway? Can you run the tests locally?
> 
> The tests are running but they aren't showing under the github repo. Check your travisci repo [page](https://travis-ci.org/github/cstjean/ScikitLearn.jl/jobs/667519138?utm_medium=notification&utm_source=github_status). The build is failing there.
> I think its not showing probably to a github authentication issue. see [issue](https://github.com/travis-ci/travis-ci/issues/9440)

@cstjean. Have to tried the recommendation on the github issue.

**Comment by cstjean** [comment]
Will do when I find some time. I'll be unavailable for the next few days. Feel free to merge when you're satisfied with your PR.

**Comment by OkonSamuel** [comment]
@cstjean. Yea. The build is now showing as github checks

File: .travis.yml [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
     allow_failures:
-    - julia: '1.0'
     - julia: nightly
 git:
     depth: 99999
 env:
     global:
     - PYTHON=conda
-before_install:
+#before_install:
```
Comment (alexmorley) [inline .travis.yml]: If you like we can delete all of these commented out lines :)

File: test/test_quickstart.jl [hunk 3]
```diff
@@ -5,14 +5,14 @@ using RDatasets: dataset
 
 iris = dataset("datasets", "iris")
 
-X = convert(Array, iris[[:SepalLength, :SepalWidth, :PetalLength, :PetalWidth]])
-y = convert(Array, iris[:Species])
+X = convert(Array, iris[!, [:SepalLength, :SepalWidth, :PetalLength, :PetalWidth]])
```
Comment (alexmorley) [inline test/test_quickstart.jl]: What are these for?

File: .travis.yml (hunk 2 above)
Comment (OkonSamuel) [reply .travis.yml]: Ok

File: test/test_quickstart.jl (hunk 3 above)
Comment (OkonSamuel) [reply test/test_quickstart.jl]: It fixes a depreciation warning from the dataframes package

**Comment by alexmorley** [comment]
:shipit:

**Pull request merged**